tests/ui/mismatched_types/int-float-type-mismatch.rs

```diff
@@ -1,7 +1,9 @@
+//! Test type mismatch error when assigning float to integer variable
```

Comment:
Nit: the original test name hints at the intention of this test I suppose -- initially, `let mut x = 2;` will cause `x` to have the inferred type `{integer}`, but this isn't "unified" yet. Still, even when not unified, a `{float}` assignment to `{integer}` clearly cannot be unified.

I would maybe describe this as

> Check that a type mismatch error is reported when trying to unify a `{float}` value assignment to an `{integer}` variable.